### PR TITLE
Various improvements to registration flow

### PR DESCRIPTION
**Pull request by robertabcd**
See commit messages.

### CODE REVIEW

File: mbbsd/register.c [hunk 1]
```diff
@@ -597,6 +597,17 @@ do_register_captcha()
...
+check_register_connection_security()
+{
+#ifdef REQUIRE_SECURE_CONN_TO_REGISTER
+    if (mbbsd_is_secure_connection())
+	return;
+    vmsg("�ШϥΦw���s�u��U�b��!");
+    exit(1);
+#endif
+}
+
```
Comment (wens) [inline mbbsd/register.c:618]: The function is only used once, and does not help with readability. Just inline it.

File: mbbsd/register.c [hunk 2]
```diff
@@ -916,27 +935,24 @@ new_register(void)
...
+    // Whether to limit login to secure connection only.
+    if (mbbsd_is_secure_connection()) {
+	// Screen full.
+	y = 17;
+	move(y, 0); clrtobot();
+	outs("[ �s�u�]�w ]");
+
+	y++;
+	if (query_yn(y, "�z�O�_�n����b���ȯ�ϥΦw���s�u�n�J? [y/n]:"))
+	    newuser.uflag |= UF_SECURE_LOGIN;
```
Comment (wens) [inline mbbsd/register.c:948]: I thought you wanted to make this the default? lol

File: mbbsd/register.c (hunk 1 above)
Comment (robertabcd) [reply mbbsd/register.c:618]: OK. Inlined.

File: mbbsd/register.c (hunk 2 above)
Comment (robertabcd) [reply mbbsd/register.c:948]: Yeah, but later I thought that user might not know what a secure connection even is... So I made them to be responsible for their own choice instead.

**Review by wens — APPROVED** [review]

**Pull request merged**